### DISCUSSION TITLE
Getting the machine license file using REST

### DISCUSSION THREAD

**Discussion by maaz-yobe** [keygen-sh/community#139] [Q&A] [answered]
Hi everyone,

I'm trying to make this sample which allows me to do activation of a machine based license online but keep a provision to deactivate the license using the offline machine license file (once TTL expires). I've activated the machine and wanted to implement an efficient method of downloading the machine license file (basically, not do it through the dashboard) upon activation of the machine. 

I wrote the following request and got the certificate but it doesn't match the contents of the file I'd get from the dashboard. Is there something I'm doing incorrectly? Really appreciate the help!

```
url = f"https://api.keygen.sh/v1/accounts/{KEYGEN_ACCOUNT_ID}/machines/{machine_fingerprint}/actions/check-out"
    headers = {
        'Authorization': f'License {license_key}',
        'Content-Type': 'application/json',
        'Accept': 'application/json'
    }
    payload = {
        'meta': {
            'include': ['license'],
            'encrypt': True
        }
    }
    response = requests.post(url, headers=headers, json=payload)
    response_data = response.json()
    print(response_data['data']['attributes']['certificate'])

````

**Comment by ezekg** [comment]
In what way does it not match? Looks like it should work to me, but that depends on what you're requesting via the dashboard. Please be more specific (with examples if possible) and I can help point you in the right direction.

**Reply by maaz-yobe** [reply]
I figured out the issue. I was also including the license in meta which was the cause of the difference in the two outputs. I am attaching the two json outputs here. My confusion is that they are giving a different TTL. Is this difference just because of the issued and expiry times being different?
[using_api_checkout.txt](https://github.com/keygen-sh/community/files/15099535/using_api_checkout.txt)
[using_dashboard_checkout.txt](https://github.com/keygen-sh/community/files/15099537/using_dashboard_checkout.txt)

Also, is there a way to specify the TTL for these files within the post request?

**Reply by ezekg** [reply, marked as answer]
Use the `meta.ttl` param to set a custom TTL. It defaults to 1 month from the current time.

**Reply by maaz-yobe** [reply]
Thank you! Really appreciate the help!
